syntax_sugar/infix.py

```diff
@@ -50,12 +50,7 @@ def step(self):
     def step(self, value):
         if value == 0 or not value /is_a/ int:
             raise TypeError('Interval must be an integer different from 0')
```

Comment:
I think we still need the code to `raise ValueError('Cannot define increasing ranges with negative step')` in order to detect potential errors early.